client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/projection/MetaProjectionTest.java

```diff
@@ -0,0 +1,66 @@
+package org.infinispan.client.hotrod.query.projection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.infinispan.configuration.cache.IndexStorage.LOCAL_HEAP;
+
+import java.util.List;
+
+import org.infinispan.client.hotrod.MetadataValue;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.Search;
+import org.infinispan.client.hotrod.test.SingleHotRodServerTest;
+import org.infinispan.commons.test.annotation.TestForIssue;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.protostream.SerializationContextInitializer;
+import org.infinispan.query.dsl.Query;
+import org.infinispan.query.dsl.QueryFactory;
+import org.infinispan.query.model.Developer;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "org.infinispan.client.hotrod.query.projection.MetaProjectionTest")
+@TestForIssue(jiraKey = "ISPN-14478")
+public class MetaProjectionTest extends SingleHotRodServerTest {
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      ConfigurationBuilder config = new ConfigurationBuilder();
+      config.indexing().enable()
+            .storage(LOCAL_HEAP)
+            .addIndexedEntity("io.dev.Developer");
+      return TestCacheManagerFactory.createServerModeCacheManager(config);
+   }
+
+   @Override
+   protected SerializationContextInitializer contextInitializer() {
+      return Developer.DeveloperSchema.INSTANCE;
+   }
+
+   @Test
+   public void testVersionProjection() {
+      RemoteCache<String, Developer> remoteCache = remoteCacheManager.getCache();
+
+      remoteCache.put("open-contributor", new Developer("iamopen", "iamopen@redmail.io", "Hibernate developer", 2000));
+      remoteCache.put("open-contributor", new Developer("iamopen", "iamopen@redmail.io", "Infinispan developer", 2000));
+      remoteCache.put("another-contributor", new Developer("mycodeisopen", "mycodeisopen@redmail.io",
+            "Infinispan engineer", 799));
+
+      MetadataValue<Developer> metadata = remoteCache.getWithMetadata("open-contributor");
+      assertThat(metadata.getVersion()).isEqualTo(2L);
+      metadata = remoteCache.getWithMetadata("another-contributor");
+      assertThat(metadata.getVersion()).isEqualTo(3L); // version is global to the cache - not to the entry
+
+      QueryFactory queryFactory = Search.getQueryFactory(remoteCache);
+      Query<Object[]> query = queryFactory.create(
+            "select d.nick, version(d), d.email, d.biography, d.contributions from io.dev.Developer d where d.biography : 'Infinispan' order by d.email");
```

Comment:
@fax4ever Would it be possible to execute a query like this:
`select d, version(d) from io.dev.Developer d where d.biography : 'Infinispan' order by d.email`

The result then would be an `array` of `Object[]` where `array[0]` would be of type `org.infinispan.query.model.Developer` and `array[1]` would be of type Long. 


Projecting all fields in the query and then building the resulting entity out of `Object[]` could bring a lot of complexity to querying.

Reply:
We don't still support entity projections along with value projections. I can try to support it. But we don't have much time to close it. I hope we can support this case within this contribution, otherwise I think we need to open a follow up.
I'll keep you posted.

Reply:
@mhajas I think I've done, I added this last commit to support this case: https://github.com/infinispan/infinispan/pull/10715/commits/5af0ee1d94f95a13a9e9eeea11131f055ae84e65

Reply:
Nice!! Thank you @fax4ever!